### PR TITLE
Reduce iOS scroll damping for lists with differently sized items

### LINKED ISSUE #32448: Scroll physics on iOS is excessively damped
I have a complex `ListView` with approximately 300 rows, and scrolling is demonstrating too much friction compared to a native list view on iOS. After some debugging, I figured out that it's because of this line in `scroll_physics.dart`:

https://github.com/flutter/flutter/blob/9c77e8e8a05a62368dbfeb05236ec2cf84d5c291/packages/flutter/lib/src/widgets/scroll_physics.dart#L355

The `createBallisticSimulation` method seems to get called several times per second during the ballistic part of scroll movements on my `ListView`, which I think is because the `ScrollMetrics.maxScrollExtent` property is changing slightly as Flutter discovers the rendered sizes of the widgets as they scroll into the viewport. However, each time it gets called, the velocity gets scaled down by `0.91`, which effectively adds a lot of additional friction to the otherwise quite smooth `BouncingScrollSimulation`.

I've worked around the issue by making my own `ScrollPhysics` class which changes the constant to `1.0` (i.e. no multiplication), and it works great, but it would be nice to have it fixed in the framework.

```
@override
  Simulation createBallisticSimulation(ScrollMetrics position, double velocity) {
    final Tolerance tolerance = this.tolerance;
    if (velocity.abs() >= tolerance.velocity || position.outOfRange) {
      return BouncingScrollSimulation(
        spring: spring,
        position: position.pixels,
        velocity: velocity * 0.91, // TODO(abarth): We should move this constant closer to the drag end.
        leadingExtent: position.minScrollExtent,
        trailingExtent: position.maxScrollExtent,
        tolerance: tolerance,
      );
    }
    return null;
  }
```

## Steps to Reproduce

Scroll any reasonably complex ListView on iOS and observe that the physics
seem to be excessively damped compared to native iOS list views.

```
[✓] Flutter (Channel master, v1.5.9-pre.193, on Mac OS X 10.14.4 18E226, locale en-US)
    • Flutter version 1.5.9-pre.193 at /Users/mil/flutter
    • Framework revision 3a6acb8c25 (8 hours ago), 2019-05-09 09:23:30 -0700
    • Engine revision 644db5a49c
    • Dart version 2.3.1 (build 2.3.1-dev.0.0 a0290f823c)


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/mil/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2.1, Build version 10E1001
    • ios-deploy 1.9.4
    • CocoaPods version 1.6.0

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 35.2.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] IntelliJ IDEA Community Edition (version 2019.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 35.3.3
    • Dart plugin version 191.7141.49

[✓] VS Code (version 1.33.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.26.0
```

### PR DESCRIPTION
**Pull request by passsy**
## Description

On iOS, lists with differently sized items have bigger damping than lists with equally sized items.

During a fling the `maxScrollExtent` continuously changes (slightly) when the items aren't equally sized. This causes a new simulation to be created. The `FastBouncingScrollPhysics` used for iOS, decreases the velocity by 9% every time it is created.
This problem wasn't tested in the existing [scroll_overlay](https://github.com/flutter/platform_tests) platform test. All items there have the same size causing the simulation to only be created once.

This PR removes the velocity change when creating the simulation and tweaks the friction of the `FrictionSimulation` to match the native iOS behavior closely for both scenarios.

The new damping isn't pixel-perfect (if that's even possible) but close. The focus of this PR is that lists with equally and unequally sized cells now behave the same on iOS.

|Before (damped)|After|
|---|---|
|damped|after|

## Related Issues

Related #11772
Fixes #32448

## Tests

For testing, I updated the [scroll_overlay PR#2](https://github.com/flutter/platform_tests/pull/2) test project to use different sized cells.

I tested with iPhone XR, iPhone SE and Pixel 4

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, this PR doesn't change the public or internal API.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_physics.dart [hunk 1]
```diff
@@ -535,9 +535,8 @@ class BouncingScrollPhysics extends ScrollPhysics {
   double get minFlingVelocity => kMinFlingVelocity * 2.0;
 
   // Methodology:
-  // 1- Use https://github.com/flutter/scroll_overlay to test with Flutter and
-  //    platform scroll views superimposed.
-  // 2- Record incoming speed and make rapid flings in the test app.
+  // 1- Use https://github.com/flutter/platform_tests/tree/master/scroll_overlay to test with
```
Comment (QuirijnGB) [inline packages/flutter/lib/src/widgets/scroll_physics.dart:538]: You missed updating the rest of the numbered list :)

Comment (passsy) [reply packages/flutter/lib/src/widgets/scroll_physics.dart:538]: Fixed, accidentally removed point 2

File: packages/flutter/lib/src/widgets/scroll_simulation.dart [hunk 2]
```diff
@@ -51,7 +51,7 @@ class BouncingScrollSimulation extends Simulation {
       _springSimulation = _overscrollSimulation(position, velocity);
       _springTime = double.negativeInfinity;
     } else {
-      _frictionSimulation = FrictionSimulation(0.135, position, velocity);
+      _frictionSimulation = FrictionSimulation(0.07, position, velocity);
```
[the remark this answers is not in the document]
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/scroll_simulation.dart:54]: The velocity we use to create the ballistic activity is responsible for the most content offset error here.

**Review by LongCatIsLooong — COMMENTED** [review]
Thank you for fixing this! And sorry for the delay, it took me longer than expected to do the research. I think we might want to keep using the `0.135` drag coefficient, but the other changes look good!

FYI the definition of breaking changes has been updated just recently, changing tests located in the test folder alone is no longer considered breaking. 

I have yet to find out what caused the difference in flinging velocity, will post my updates in https://github.com/flutter/flutter/issues/11772.

**Comment by goderbauer** [comment]
(PR triage) @LongCatIsLooong @passsy What is the status of this PR? Should it be closed in favor of https://github.com/flutter/flutter/pull/60501?

**Comment by passsy** [comment]
@LongCatIsLooong I reverted the drag from `FrictionSimulation` back to `0.135`. This only resolves the excessive damping when list items have different sizes. It now scrolls - faster - like lists with equally sized items.

|Before (damped)|WIP `drag=0.07`|After `drag=0.135`|
|---|---|---|
|damped|after|after2|

#60501 does not fix the issue here where recreating the simulation reduces the scroll speed. Both PRs improve different parts of the iOS scrolling behaviour.

**Comment by LongCatIsLooong** [comment]
@goderbauer @passsy I didn't add tests in #60501 to cover the case where the viewport needs to apply a new content dimension. I was hoping that we can wait for both PRs to be reviewed and then merge them together.

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. Thank you for fixing this long-standing issue (and updating the `scroll_overlay` app)! I'll try to merge this PR and #60501 together to prevent potential regression. 
You may need to merge the latest `upstream/master` to the PR branch to fix the "docs-linux" CI failure.

**Pull request merged**